pkg/cpi/commonUtils.go

```diff
@@ -65,3 +61,10 @@ func (tokenParameters TokenParameters) GetBearerToken() (string, error) {
 	token := jsonResponse.Path("access_token").Data().(string)
 	return token, nil
 }
+
+//CloseResponseBodyIfNecessary - close http response object
+func CloseResponseBodyIfNecessary(response *http.Response) {
+	if response != nil && response.Body != nil {
+		response.Body.Close()
+	}
```

Comment:
As @Marcus pointed out: [it is best to check the if the response and the response body are nil before the defer.](https://github.com/SAP/jenkins-library/pull/2474#issuecomment-756699583)